### ISSUE TITLE
Accessibility Feedback for Midpoint Review [Healthcare Experience, Check-in MVP]

### ISSUE THREAD

**Issue by joshkimux** [department-of-veterans-affairs/va.gov-team#26325]
# Accessibility Feedback for Midpoint Review [Healthcare Experience, Check-in MVP]

## Feedback framework

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

## Point of Contact

**VFS Point of Contact:** Josh

## Must
- Manage focus page to page to the `h1` using a `tabindex="-1"`
- Use a `fieldset` and `legend` for the yes and no buttons so that the question is announced to screen reader users

## Should
- "We'll let you know..." heading should not be a heading if it has no additional content as it's not starting a group; instead, use a `p` and link it to the `h1` using `aria-describedby`. Alternatively, if more helpful context can be provided (e.g. _how_ someone might let them know when they're ready) "We'll let you know..." should be a `h2` not a `h3`

<img width="1407" alt="Screen Shot 2021-06-22 at 1 47 03 PM" src="https://user-images.githubusercontent.com/14154792/122975152-0e4abe80-d361-11eb-89f3-aa67373922b3.png">

- If more pages are added to the form, the yes/no buttons should be converted to radio options to accommodate a persistent `continue` button
- Provide users a method of moving backwards in case they selected the wrong choice e.g. fat fingering "yes" when you meant to select "no"

## Consider
- Providing hint text examples of insurance, contact, or other information
- Providing more instructions under "please check in with a staff member"

<hr aria-hidden="true"/>

## Fieldset and legend for yes and no buttons
Screen readers will not hear the instructions associated with each of the buttons unless:
- the `h1` is nested within a `legend` 
- the `legend` is the first child of a `fieldset` that wraps around all the form elements

### Before

https://user-images.githubusercontent.com/14154792/122978049-10fae300-d364-11eb-8f7a-b92529405a59.mov

### After

https://user-images.githubusercontent.com/14154792/122977983-ff194000-d363-11eb-82b7-2c7e0fd81326.mov

<hr aria-hidden="true"/>

## Confirmation page option examples
Designer to determine what works best to meet intent

### A. `h2` is now a `p` that describes the `h1` using `aria-describedby`

https://user-images.githubusercontent.com/14154792/122976198-1fe09600-d362-11eb-9e6e-cb6e6d599d2d.mov

### B. `h1` is included within the alert itself and the `h2` is now a `p` that's a child to the `h1`

https://user-images.githubusercontent.com/14154792/122976630-91204900-d362-11eb-8630-253b6371f7e9.mov

### C. Additional details are provided below the `h2` justifying it as a `h2`

https://user-images.githubusercontent.com/14154792/122977245-2c192300-d363-11eb-8256-d7ed42963785.mov

**Comment by mdewey** [comment]
The `MUSTS` were addressed in  [this PR](https://github.com/department-of-veterans-affairs/vets-website/pull/17734).

**Comment by joshkimux** [comment]
@mdewey Fieldset is working well!

As for focus management it looks like you've added a `tabindex="-1"` but you may be missing a script to send the focus to the `h1`. `tabindex="-1"` allows an element to receive focus, but doesn't actually give it focus. You'll need to use javascript to do so.

**Comment by mdewey** [comment]
Good Morning @joshkimux! I have updated the focus for our app. Can you check it out, I think its what it should, but I am not 100% sure. 

Let me know what you think :-)

**Comment by joshkimux** [comment]
@mdewey Is there an updated link I can use to check it out? The one I've been using is no longer working 😞

**Comment by mdewey** [comment]
Absolutely

https://staging.va.gov/health-care/appointment-check-in/?id=46bebc0a-b99c-464f-a5c5-560bc9eae287

**Comment by joshkimux** [comment]
<img width="1463" alt="Screen Shot 2021-07-13 at 11 58 17 AM" src="https://user-images.githubusercontent.com/14154792/125485303-ee50689d-20ae-4040-ac86-c38dd62ed4be.png">

Can confirm focus is working!! I would recommend removing the `aria-label` from your `h1`s.  [Please reference the suggestions I left in the original ticket on using `aria-describedby` or heading structures](https://github.com/department-of-veterans-affairs/va.gov-team/issues/26325#Confirmation-page-option-examples) 🙇

**Comment by julieadair1** [comment]
@jparker0425 is there still work to do on this ticket?

**Comment by jparker0425** [comment]
@julieadair1 this can be closed out. This was done prior to the MVP being released and there have been several changes since then.

cc: @mdewey

**Comment by julieadair1** [comment]
@jparker0425 , perfect. thanks. closing this ticket.
